third_party/pub/batch_analyze_licenses.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:pana/src/license.dart';
+
+Future<void> main() async {
```

Comment:
Write a comment that says what this tool does....
```suggestion
Future<void> main(List<String> args) async {

  if (args.length > 1 && args[1] == '--help') {
    print('''
Run license detection on all files matching `.dart_tool/pana/license-cache/LICENSE-<package>`.

Creates a file named: `detected-licenses.<date-time>.json`, on the form:
{
  "<package>": {
    "spdxIds": ["<detected SPDX ID>", ...]
  },
  ...
},
''');
  }
```

Supposing we change where it writes files...